pkg/workflow/workflowrun/operator.go

```diff
@@ -454,10 +456,37 @@ func (o *operator) GC(lastTry, wfrDeletion bool) error {
 				o.recorder.Eventf(o.wfr, corev1.EventTypeWarning, "GC", "Delete pod '%s' error: %v", status.Pod.Name, err)
 			}
 		} else {
-			log.WithField("ns", status.Pod.Namespace).WithField("pod", status.Pod.Name).Info("Pod deleted")
+			log.WithField("ns", status.Pod.Namespace).WithField("pod", status.Pod.Name).Info("Start to delete pod")
+
+			wg.Add(1)
+			go func(namespace, podName string) {
+				defer wg.Done()
+
+				timeout := time.After(5 * time.Minute)
+				ticker := time.NewTicker(5 * time.Second)
+				defer ticker.Stop()
+				for {
+					select {
+					case <-timeout:
+						log.WithField("ns", namespace).WithField("pod", podName).Warn("Pod deletion timeout")
+						return
+					case <-ticker.C:
+						_, err := o.clusterClient.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+						if err != nil && errors.IsNotFound(err) {
+							log.WithField("ns", namespace).WithField("pod", podName).Info("Pod deleted")
+							return
+						}
```

Comment:
Miss the logic of https://github.com/caicloud/cyclone/pull/1321/files#diff-9bb1eecfaf9ae747dcffe86457ab7f58R450-R457

Any way to merge the first deletion and the retries.

Reply:
I think it's not necessary, since if deleting pod gets result `error not found`, the for loop will continue.